### ISSUE TITLE
cannot connect to replicaset over SSH tunnels

### ISSUE THREAD

**Issue by datanator** [mongodb/mongo-php-library#131]
I want to connect from my development environment to a mongo replicaset (3 x mogod, 1 master).

This will not work. Though not a single exception or error output is shown while connecting or selecting db and collection, it will just hang whenever I want to execute a (find) query.
Connecting to the master instance not using replicasets does work!

So I tried pushing all option permutations that made sense and also tried to push server fqdns in my windows hosts file: nothing seems to respond.

I found that in the old PHP mongo driver it was possible to check connections to the replicaset, in the current mongodb driver + library this seems no longer possible hence debugging is very difficult.

I use PHP 5.6.9 with mongodb 1.1.2 driver running on windows 8 to a MongoDB 2.6.11 server running on Ubuntu 3.2

Cannot find any comments on this on the internet.

```
require_once __DIR__ . "/vendor/autoload.php";

// note that the ipnumbers designate ssh tunnes that are working for Umongo, 
// or when using non replicaset urls
$url = "mongodb://192.168.x.x:27021,192.168.x.x:27022,192.168.x.x:27023/";
$opt = array("connectTimeoutMS" => 10000,"replicaSet"=> "setname", "connect" => true);
$client = new MongoDB\Client($url, $opt);
$collection = $client->selectCollection('mydb', 'mycol');
// this line gets executed
$result = $collection->find(array());
// this line never gets reached
foreach($result as $entry) {
    var_dump($entry);
}
```

**Comment by jmikola** [comment]
> I found that in the old PHP mongo driver it was possible to check connections to the replicaset, in the current mongodb driver + library this seems no longer possible hence debugging is very difficult.

The new extension actually provides more visibility and access to the connected nodes than the legacy driver. [`Manager::getServers()`](http://php.net/manual/en/mongodb-driver-manager.getservers.php) returns a list of Server objects, from which you can access properties of the node and the most recent [isMaster](https://docs.mongodb.org/manual/reference/command/isMaster/) response. The [`Manager::selectServer()`](http://php.net/manual/en/mongodb-driver-manager.selectserver.php) method can also be used to select an individual server, given a read preference.

> `$opt = array("connectTimeoutMS" => 10000,"replicaSet"=> "setname", "connect" => true);`

The `connect` option only existed in the legacy driver and can be removed here. The current extension, which wraps [libmongoc](https://github.com/mongodb/mongo-c-driver), supports only the standard [connection string options](https://docs.mongodb.org/manual/reference/connection-string/).

I can suggest two things to look into for further diagnosis:

 * Assuming you're seeing the PHP script time out before the driver responds, it's possible that your execution limit is less than the driver's timeout for server selection. Similar to how you've configured `connectTimeoutMS` (i.e. for opening sockets) in the client options, there is another related option called [`serverSelectionTimeoutMS`](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#serverselectiontimeoutms). I would suggest lowering that from its 30-second default to see if you get an exception.

 * The extension exposes a [mongodb.debug](http://php.net/manual/en/mongodb.configuration.php#ini.mongodb.debug) INI option, which you can use to log all internal driver activity to a directory, file, or stderr. If you'd like to set that for a small test script and share the output, I'd be happy to try and make sense of it for you.

Note: I realize that `serverSelectionTimeoutMS` (and related options from that linked specification) are not documented in the server's manual page for [connection string options](https://docs.mongodb.org/manual/reference/connection-string/). I've opened [DOCS-7416](https://jira.mongodb.org/browse/DOCS-7416) to track that improvement.

**Comment by jmikola** [comment]
Closing this due to inactivity. If you'd like to follow-up, please attempt reproducing your issue with the recently released [1.2.0](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.0) extension, as that includes significant changes to our connection handling.
